src/types/expiration_time.ts

```diff
@@ -0,0 +1,74 @@
+// External libraries
```

Comment:
ExpirationTime feels like something I'd initialize with a specific time. I.e., if I wanted a loan to expire on December 31st, 2018, I'd do 

`new ExpirationTime("December 31st, 2018);`

The use case you implement here feels more like a `ValidityPeriod` (bad name) or something else...

`ValidityPeriod(2, "weeks");`
`ExpirationTime.ValidFor(2, "weeks");`